src/scripts/main.js

```diff
@@ -1,3 +1,199 @@
 'use strict';
 
-// write your code here
+const startButton = document.querySelector('.start');
+const startMessage = document.querySelector('.message-start');
+const loseResult = document.querySelector('.message-lose');
+const winResult = document.querySelector('.message-win');
+const gameScore = document.querySelector('.game-score');
+const columns = [[], [], [], []];
+const rows = [...document.querySelectorAll('.field-row')];
+const cells = document.querySelectorAll('.field-cell');
+const cellsInRow = rows.map((row) => [...row.querySelectorAll('.field-cell')]);
+const cellStartValues = [2, 2, 2, 2, 2, 2, 2, 2, 2, 4];
+
+for (let i = 0; i < 4; i++) {
+  for (let k = 0; k < 4; k++) {
+    columns[k].push(rows[i].children[k]);
+  }
+}
+
+let clearCells = [...cells];
+let started = false;
+let rotated = false;
+
+function randomCellFill() {
+  if (!clearCells.length) {
+    return;
+  }
+
+  const randomIndex = (someArr) => Math.floor(Math.random() * someArr.length);
+  const randomCell = clearCells[randomIndex(clearCells)];
+  const randomValue = cellStartValues[randomIndex(cellStartValues)];
+
+  randomCell.classList = (`field-cell field-cell--${randomValue}`);
+  randomCell.innerText = randomValue;
+
+  clearCells.splice(clearCells.indexOf(randomCell), 1);
+}
+
+const start = () => {
+  startMessage.style = 'display: none;';
+  startButton.classList = 'button restart';
+  startButton.innerText = 'Restart';
+
+  randomCellFill();
+  randomCellFill();
+  started = true;
+};
+
+const restart = () => {
+  loseResult.classList.add('hidden');
+  winResult.classList.add('hidden');
+  clearCells = [...cells];
+  gameScore.innerText = 0;
+
+  cells.forEach(cell => {
+    cell.innerText = '';
+    cell.className = 'field-cell';
+  });
+
+  randomCellFill();
+  randomCellFill();
+};
+
+function isMergePossible() {
+  const wholeField = [...cellsInRow, ...columns];
+
+  for (const line of wholeField) {
+    for (let i = 0; i < 3; i++) {
+      if (line[i].innerText === line[i + 1].innerText) {
+        return true;
+      }
+    }
+  }
+}
+
+function cellUp() {
+  for (const column of columns) {
+    rotateCells([...column].reverse());
+  }
+}
+
+function cellDown() {
+  for (const column of columns) {
```

Comment:
Avoid using a `for...of` loop to calculate the sum of an array. Instead, use `forEach`

Reply:
do it in all cases